src/modules/social/posts/posts.route.ts

```diff
@@ -0,0 +1,130 @@
+import { FastifyInstance } from "fastify"
+
+import { $ref } from "./posts.schema"
+import {
+  getPostsHandler,
+  getPostHandler,
+  createPostHandler,
+  updatePostHandler,
+  reactionHandler
+} from "./posts.controller"
+
+async function postsRoutes(server: FastifyInstance) {
+  server.get(
+    "/",
+    {
+      preHandler: [server.authenticate],
+      schema: {
+        tags: ["posts"],
+        security: [{ bearerAuth: [] }],
+        response: {
+          200: {
+            type: "array",
+            items: $ref("displayPostSchema")
+          }
+        }
+      }
+    },
+    getPostsHandler
+  )
+
+  server.post(
+    "/",
+    {
+      preHandler: [server.authenticate],
+      schema: {
+        body: $ref("createPostBaseSchema"),
+        tags: ["posts"],
+        security: [{ bearerAuth: [] }],
+        response: {
+          200: $ref("postSchema")
+        }
+      }
+    },
+    createPostHandler
+  )
+
+  server.put(
+    "/:id",
+    {
+      preHandler: [server.authenticate],
+      schema: {
+        body: $ref("createPostSchema"),
+        tags: ["posts"],
+        security: [{ bearerAuth: [] }],
+        response: {
+          200: $ref("postSchema")
+        }
+      }
+    },
+    updatePostHandler
+  )
+
+  server.delete(
+    "/:id",
+    {
+      preHandler: [server.authenticate],
+      schema: {
+        params: {
+          type: "object",
+          properties: {
+            id: { type: "integer" }
+          }
+        },
+        tags: ["posts"],
+        security: [{ bearerAuth: [] }],
+        response: {
+          200: {
+            type: "object"
+          }
+        }
+      }
+    },
+    updatePostHandler
+  )
+
+  server.get(
+    "/:id",
+    {
+      preHandler: [server.authenticate],
+      schema: {
+        tags: ["posts"],
+        security: [{ bearerAuth: [] }],
+        params: {
+          type: "object",
+          properties: {
+            id: { type: "integer" }
+          }
+        },
+        response: {
+          200: $ref("displayPostSchema")
+        }
+      }
+    },
+    getPostHandler
+  )
+
+  server.get(
```

Comment:
Are you using GET here on purpose? Seeing as it performs a db update would it not be better suited for PUT/PATCH?

Reply:
Convenience is the best answer. It could be a POST or a PATCH depending on if you are adding a new reaction or updating the count of an existing one

Reply:
Honeslty I'm fine with either. Just stood out to me when looking through.